File: backend/app/schemas/notification.py
"""站内通知响应 schema（Phase 5A）。"""

from __future__ import annotations

from datetime import datetime
from typing import Any, Literal

from pydantic import BaseModel, Field


class NotificationRead(BaseModel):
    id: str
    type: str
    entity_type: str | None
    entity_id: str | None
    params: dict[str, Any]
    actor_user_id: str | None
    is_read: bool
    read_at: datetime | None
    created_at: datetime


class UnreadCount(BaseModel):
    count: int


class ReadAllResult(BaseModel):
    updated: int


class PushTokenRegister(BaseModel):
    token: str = Field(min_length=1, max_length=512)
    platform: Literal["ios", "android", "web"]


class PushTokenDelete(BaseModel):
    token: str = Field(min_length=1, max_length=512)


class PushTokenRead(BaseModel):
    id: str
    token: str
    platform: str

File: frontend/src/types/notification.ts
export interface Notification {
  id: string
  type: string
  entity_type: string | null
  entity_id: string | null
  params: Record<string, unknown>
  actor_user_id: string | null
  is_read: boolean
  read_at: string | null
  created_at: string
}

export interface NotificationPreference {
  email_enabled: boolean
  disabled_types: string[]
}

export interface ListNotificationsParams {
  page?: number
  page_size?: number
  is_read?: boolean
  type?: string
}
